pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -59,76 +59,80 @@ contract WeightedPool is BaseWeightedPool, InvariantGrowthProtocolFees {
     uint256 internal immutable _normalizedWeight6;
     uint256 internal immutable _normalizedWeight7;
 
+    struct NewPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        address[] assetManagers;
+        uint256 swapFeePercentage;
```

Comment:
What do you think about adding the owner to the struct?

In ComposableStable, we had to put everything into one struct because of the stack, but here I think we have a choice. I prefer it like this, where only the "parameters" that can be set by the owner/manager are in the struct. The owner itself is one of those params, though.

Reply:
I aimed for consistency with [ManagedPool](https://github.com/balancer-labs/balancer-v2-monorepo/blob/381e8318776c6dbb92c5e2192ecdc13df3ef3d5e/pkg/pool-weighted/contracts/smart/ManagedPool.sol#L166-L178) when creating this struct so I left it out to match that.

Moving it inside gives the benefit of us being able to pass the struct and only the struct to the factory (we have a small bit of inconsistency in the factories atm as ManagedPool takes a struct and owner while WeightedPool takes individual args)

Reply:
Ah, we leave `owner` outside of the struct as when we have a pool controller then it's not a user chosen value. We'd then have to have two different structs or pass a dummy value for the owner.

Reply:
> while WeightedPool takes individual args

and this is due to asset managers being non-optional in one of the factories.